packages/ui/__tests__/Tooltip.test.tsx

```diff
@@ -1,64 +1,107 @@
 import React from 'react'
+import { act } from 'react-dom/test-utils'
 import { mount } from 'enzyme'
-import RCTooltip from 'rc-tooltip'
 
-import { Tooltip, TooltipProps } from '../src/Tooltip'
-
-import styles from '../src/Tooltip.module.scss'
+import { Tooltip } from '../src/Tooltip'
 
+/**
+ * There's a weird issue with react-popper that results in missing act() wrapper error.
+ * This is why all mounts are wrapped in act().
+ *
+ * https://github.com/popperjs/react-popper/issues/368
+ */
 describe('Tooltip', () => {
-  test('renders children without Tooltip if "overlay" is undefined', () => {
-    const Child = () => <div />
+  test('Renders correctly if "content" property is defined.', async () => {
+    await act(async () => {
+      const wrapper = mount(
+        <Tooltip id="tooltip-test" content="Tooltip content">
+          {(ref) => (
+            <button ref={ref} data-test="tooltip-reference">
+              Hover me
+            </button>
+          )}
+        </Tooltip>,
+      )
+
+      const tooltipOverlay = wrapper.find('[data-test="tooltip-overlay"]')
+
+      expect(wrapper.find('[data-test="tooltip-reference"]').exists()).toBeTruthy()
+      expect(tooltipOverlay.exists()).toBeTruthy()
+      expect(tooltipOverlay.text()).toEqual('Tooltip content')
+    })
+  })
+
+  test('Does not render tooltip if "content" property is not defined.', async () => {
+    await act(async () => {
+      const wrapper = mount(
+        <Tooltip id="tooltip-test" content={undefined}>
+          {(ref) => (
+            <button ref={ref} data-test="tooltip-reference">
+              Hover me
+            </button>
+          )}
+        </Tooltip>,
+      )
+
+      expect(wrapper.find('[data-test="tooltip-overlay"]').exists()).toBeFalsy()
+
+      wrapper.find('[data-test="tooltip-reference"]').simulate('mouseenter')
+
+      wrapper.update()
+
+      expect(wrapper.find('[data-test="tooltip-overlay"]').exists()).toBeFalsy()
+    })
+  })
+
+  test('Shows tooltip overlay on hover of the target element.', async () => {
+    await act(async () => {
+      const wrapper = mount(
+        <Tooltip id="tooltip-test" content="Tooltip content">
+          {(ref) => (
+            <button ref={ref} data-test="tooltip-reference">
+              Hover me
+            </button>
+          )}
+        </Tooltip>,
+      )
+
+      let tooltipOverlay = wrapper.find('[data-test="tooltip-overlay"]')
+
+      expect(tooltipOverlay.hasClass('hidden')).toBeTruthy()
+
+      wrapper.find('[data-test="tooltip-reference"]').simulate('mouseenter')
 
-    const wrapper = mount(
-      <Tooltip overlay={undefined}>
-        <Child />
-      </Tooltip>,
-    )
+      wrapper.update()
 
-    expect(wrapper.exists(RCTooltip)).toBeFalsy()
-    expect(wrapper.exists(Child)).toBeTruthy()
+      tooltipOverlay = wrapper.find('[data-test="tooltip-overlay"]')
+
+      expect(tooltipOverlay.hasClass('hidden')).toBeTruthy()
```

Comment:
Shouldn't we expect it to be `false` here?

Reply:
I had to remove the entire test on hover because for some reason I couldn't get `simulate('mouseenter')` to work. I search for a solution for a long time, but it seems like this is an issue with Enzyme. If you have any suggestion it would be really appreciated!

Reply:
@jozefhruska I might know what's the problem there. Let me test a possible solution and get back to you!

Reply:
e@jozefhruska So, problem with `simulate` is that it maps given string to a react handler function (`'change' -> 'onChange'`, `'mouseenter' -> 'onMouseEnter'`, ...) and tries to trigger these function on given wrapper.

I was thinking about mocking `addEventListener` and saving the listener passed to it to local variable and then calling it as needed. Id didn't work out, I was only able to do it with `document`'s and `window`'s `addEventListener` function but not with a `addEventListener` bound to an element.